spot_driver_cpp/src/api/default_kinematic_api.cpp

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) 2023 Boston Dynamics AI Institute LLC. All rights reserved.
+
+#include <spot_driver_cpp/api/default_kinematic_api.hpp>
+
+namespace spot_ros2 {
+DefaultKinematicApi::DefaultKinematicApi(bosdyn::client::InverseKinematicsClient* kinematic_client)
+    : kinematic_client_{kinematic_client} {}
+
+tl::expected<InverseKinematicsResponse, std::string> DefaultKinematicApi::getSolutions(
+    InverseKinematicsRequest& request) {
+  try {
+    auto result = kinematic_client_->InverseKinematics(request);
+    if (result) {
+      return result.response;
+    }
+    return tl::make_unexpected(result.status.message());
```

Comment:
Related to my other comment about empty error messages, I think the `message()` field isn't actually filled in by the Spot API in all cases. If we want to add some useful output here we might need to read the error code and have a switch statement that prints out an explanation for each error code.

Alternatively, there's also a debug string field which might contain useful info.

Generally I think it's beneficial to the user to provide as much detail as possible about what went wrong in this case.

Reply:
Not knowing ahead of time which error code can be returned, I decided to return a message containing both error code and possible optional message.